packages/netlify-cms-core/src/lib/registry.js

```diff
@@ -78,25 +80,32 @@ export function registerWidget(name, control, preview) {
       name: widgetName,
       controlComponent: control,
       previewComponent: preview,
+      allowMapValue,
       globalStyles,
+      ...options
     } = name;
     if (registry.widgets[widgetName]) {
-      console.error(oneLine`
+      console.warn(oneLine`
         Multiple widgets registered with name "${widgetName}". Only the last widget registered with
         this name will be used.
       `);
     }
     if (!control) {
       throw Error(`Widget "${widgetName}" registered without \`controlComponent\`.`);
     }
-    registry.widgets[widgetName] = { control, preview, globalStyles };
+    registry.widgets[widgetName] = { control, preview, globalStyles, allowMapValue, ...options };
```

Comment:
Widgets can choose to allow their values to be maps (like object widget does), and also pass in configurable values the CMS doesn't know about.